google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/appengine/util/AppEngineUtilTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2016 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.util;
+
+import static org.junit.Assert.assertEquals;
+
+import org.joda.time.DateTimeUtils;
+import org.junit.After;
+import org.junit.Test;
+
+/**
+ * Tests App Engine Utilities
+ * Uses an external trusted datetime to milli converter to set the datetime to the desired value.
+ */
+public class AppEngineUtilTest {
+
+  @Test
+  public void testDateTimeToVersion() {
```

Comment:
To make this work properly I think you'll have to make an
AppEngineVersionGenerator(Clock)  where for tests you can provide a mock
Clock that returns the instant that specifies some fixed point in time you
can test against.  Or actually I think you can pass a fixed time Clock as
well.

On Wed, Apr 6, 2016 at 2:29 PM, Etan Shaul <notifications@github.com> wrote:

> In
> google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/appengine/util/AppEngineUtilTest.java
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/603#discussion_r58757241>
> :
>
> > +package com.google.cloud.tools.intellij.appengine.util;
> > +
> > +import static org.junit.Assert.assertEquals;
> > +
> > +import org.joda.time.DateTimeUtils;
> > +import org.junit.After;
> > +import org.junit.Test;
> > +
> > +/**
> > + * Tests App Engine Utilities
> > + * Uses an external trusted datetime to milli converter to set the datetime to the desired value.
> > + */
> > +public class AppEngineUtilTest {
> > +
> > +  @Test
> > +  public void testDateTimeToVersion() {
>
> Could you explain a bit? If I use a clock internally with real time then
> the test feels a bit flaky to me because it depends on when the test is
> run. If the clock is set to a preconfigured datetime then the problem is
> circular because it would require using a formatter to set it (which is
> what we are testing in the first place). I tried mocking DateTime but
> didn't have much luck with that approach.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/603/files/9008c7a4ca0958685f37f59f4cd92c80e2ed9097#r58757241>
>